plugins/woocommerce/tests/php/src/Internal/Admin/Settings/PaymentProviders/WooPayments/WooPaymentsServiceTest.php

```diff
@@ -0,0 +1,4903 @@
+					'is_store_connected'  => true,
+					'has_connected_owner' => false,
+				),
+				array(
+					'has_account'       => true,
+					'has_valid_account' => true,
+					'test_account'      => false,
+				),
+			),
+			'business_verification - stored started and completed with valid live account, met requirements' => array(
+				WooPaymentsService::ONBOARDING_STEP_BUSINESS_VERIFICATION,
+				WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED,
+				array(
+					WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED => time() - 10,
+					WooPaymentsService::ONBOARDING_STEP_STATUS_COMPLETED => time(),
+				),
+				array(
+					'is_store_connected'  => true,
+					'has_connected_owner' => true,
+				),
+				array(
+					'has_account'       => true,
+					'has_valid_account' => true,
+					'test_account'      => false,
+				),
+			),
+		);
+	}
+
+	/**
+	 * Test that set_onboarding_step_started throws an exception when an invalid step ID is provided.
+	 *
+	 * @return void
+	 */
+	public function test_set_onboarding_step_started_throws_on_invalid_step_id() {
+		$location = 'US';
+
+		$this->expectException( \Exception::class );
+		$this->expectExceptionMessage( 'Invalid onboarding step ID' );
+
+		$this->sut->set_onboarding_step_started( 'invalid_step_id', $location );
+	}
+
+	/**
+	 * Test that set_onboarding_step_started throws an exception when the requirements are not met for the step.
+	 *
+	 * @return void
+	 */
+	public function test_set_onboarding_step_started_throws_on_unmet_requirements() {
+		$location = 'US';
+
+		// Arrange the WPCOM connection.
+		$this->mock_wpcom_connection_manager
+			->expects( $this->any() )
+			->method( 'is_connected' )
+			->willReturn( true );
+		$this->mock_wpcom_connection_manager
+			->expects( $this->any() )
+			->method( 'has_connected_owner' )
+			->willReturn( false );
+
+		$this->expectException( \Exception::class );
+		$this->expectExceptionMessage( 'Onboarding step can no be started because requirements are not met.' );
+
+		$this->sut->set_onboarding_step_started( WooPaymentsService::ONBOARDING_STEP_TEST_ACCOUNT, $location );
+	}
+
+	/**
+	 * Test that set_onboarding_step_started does not overwrite the existing step status if the overwrite flag is false.
+	 *
+	 * @return void
+	 */
+	public function test_set_onboarding_step_started_does_not_overwrite() {
+		$location = 'US';
+
+		// Arrange.
+		$step_id                = WooPaymentsService::ONBOARDING_STEP_PAYMENT_METHODS;
+		$timestamp              = time() - 100;
+		$stored_profile         = array(
+			'onboarding' => array(
+				$location => array(
+					'steps' => array(
+						$step_id => array(
+							'statuses' => array(
+								WooPaymentsService::ONBOARDING_STEP_STATUS_STARTED => $timestamp,
+							),
+						),
+					),
+				),
+			),
+		);
+		$updated_stored_profile = array();
+		$this->mockable_proxy->register_function_mocks(
+			array(
+				'get_option'    => function ( $option_name, $default_value = null ) use ( $stored_profile ) {
```

Comment:
I think this and update_option could be positioned in init because it's getting duplicated?

Reply:
Nope. Can't do that because the `get_option` and `update_option` closures need to use the test mocked stored profile variable (see `use ( $stored_profile )`). It would get quite messy if we would use a test class property passed by reference, etc.